### PR TITLE
Feat color collection

### PR DESCRIPTION
**Pull request by pauliuskalv**

### CODE REVIEW

File: TOPBanga/Detection/ColorDetector.cs [hunk 1]
```diff
@@ -7,51 +7,61 @@
...
 
+        public Bgr circleColor { get; set; }
+
+        public int circleWidth { get; set; }
+
+
         public ColorDetector()
         {
-            this.threshold = 35;
+            this.threshold = 18; // default threshold
```
Comment (mrzilvis) [inline TOPBanga/Detection/ColorDetector.cs]: why 18 and not 35 anymore?

File: TOPBanga/Detection/ColorDetector.cs [hunk 2]
```diff
@@ -7,51 +7,61 @@
...
-        public void SetBallColorHSV(int h, int s, int v)
+        public ColorDetector(int threshold)
         {
-            this.ballHsv = new Hsv(h, s, v);
+            this.threshold = threshold;
         }
 
-        public bool DetectBall(out float x, out float y, out float radius, out Bitmap bitmap, int minRadius = 1)
+        public bool DetectBall(out float x, out float y, out float radius, out Bitmap bitmap, Hsv ballHsv, int minRadius = 1,
+            int cannyThreshold = 12, int accumulatorThreshold = 26, double resolution = 1.9, double minDist = 10, int HoughMinRadius = 0, int HoughMaxRadius = 0)
```
Comment (mrzilvis) [inline TOPBanga/Detection/ColorDetector.cs]: Separate into more lines

File: TOPBanga/VideoFromFile.cs [hunk 3]
```diff
@@ -43,17 +48,22 @@ private void BrowseButton_Click(object sender, EventArgs e)
...
-            int y = mouseEventArgs.Y;
-            this.detector.SetBallColorHSVFromCoords(x, y);
-            Image<Hsv, byte> colorImage = new Image<Hsv, byte>(25, 25, this.detector.ballHsv);
-            this.ColorBox.Image = colorImage.Bitmap;
+            if (this.colorNeeded)
+            {
+                MouseEventArgs mouseEventArgs = (MouseEventArgs)e;
+                int x = mouseEventArgs.X;
+                int y = mouseEventArgs.Y;
+                colorContainer.Add(this.detector.GetBallColorHSVFromCoords(x, y));
```
Comment (mrzilvis) [inline TOPBanga/VideoFromFile.cs]: I think this doesn't update color in the colorBox.

File: TOPBanga/Detection/ColorDetector.cs (hunk 1 above)
Comment (pauliuskalv) [reply TOPBanga/Detection/ColorDetector.cs]: We can reduce the range of colors, because we can always prompt to select a new color, hence the lower value

**Pull request merged**